x-pack/plugins/reporting/public/lib/reporting_api_client/reporting_api_client.ts

```diff
@@ -225,6 +227,8 @@ export class ReportingAPIClient implements IReportingAPI {
   }
 
   public migrateReportingIndicesIlmPolicy() {
-    return this.http.put(INTERNAL_ROUTES.MIGRATE.MIGRATE_ILM_POLICY);
+    return this.config.statefulSettings.enabled
+      ? this.http.put(INTERNAL_ROUTES.MIGRATE.MIGRATE_ILM_POLICY)
+      : null;
```

Comment:
Let's just try to make sure this method is not called in serverless. Then, we wouldn't need to pass the `config` object or change this class at all.

In `x-pack/plugins/reporting/public/management/report_listing.tsx`, the line, `<MigrateIlmPolicyCallOut toasts={toasts} />` gets us into the code path where this is called. We should update `report_listing.tsx` to keep us out of the code path by using the config in that file in a conditional:


```diff
           }
         />
 
-        <MigrateIlmPolicyCallOut toasts={toasts} />
+        {config.statefulSettings.enabled ? <MigrateIlmPolicyCallOut toasts={toasts} /> : null}
 
         <EuiSpacer size={'l'} />
         <div>{this.renderTable()}</div>
```